be/src/vec/columns/column_nullable.cpp

```diff
@@ -390,17 +391,23 @@ int ColumnNullable::compare_at(size_t n, size_t m, const IColumn& rhs_,
     /// instead of the usual nan_direction_hint and is used to implement
     /// the ordering specified by either NULLS FIRST or NULLS LAST in the
     /// ORDER BY construction.
-    const auto& nullable_rhs = assert_cast<const ColumnNullable&>(rhs_);
```

Comment:
The way change the default behavior in Doris， `compare_at` must use same type column